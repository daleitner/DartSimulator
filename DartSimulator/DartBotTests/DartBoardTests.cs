using System.Linq;
using ApprovalTests;
using ApprovalTests.Reporters;
using DartBot;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace DartBotTests
{
	[TestClass]
	[UseReporter(typeof(DiffReporter))]
	public class DartBoardTests
	{
		[TestMethod]
		public void DartBoardShouldHaveGotAllFields()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields = dartBoard.Fields.Select(x => x.ToString());
			Approvals.VerifyAll(fields, "Field");
		}

		[TestMethod]
		public void GetDoubleBullShouldReturnDoubleBullField()
		{
			var dartBoard = DartBoard.GetInstance();
			var doubleBull = dartBoard.GetDoubleBull();
			Approvals.Verify(doubleBull);
		}

		[TestMethod]
		public void GetSingleBullShouldReturnSingleBullField()
		{
			var dartBoard = DartBoard.GetInstance();
			var singleBull = dartBoard.GetSingleBull();
			Approvals.Verify(singleBull);
		}

		[TestMethod]
		public void GetSingleInFieldShouldReturnSingleInFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.SingleIn).Select(x => dartBoard.GetSingleInField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}
		[TestMethod]
		public void GetSingleOutFieldShouldReturnSingleOutFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.SingleOut).Select(x => dartBoard.GetSingleOutField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}
		[TestMethod]
		public void GetDoubleFieldShouldReturnDoubleFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.Double).Select(x => dartBoard.GetDoubleField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}
		[TestMethod]
		public void GetTripleFieldShouldReturnTripleFieldWithRightValue()
		{
			var dartBoard = DartBoard.GetInstance();
			var fields =
				dartBoard.Fields.Where(x => x.Type == FieldEnum.Triple).Select(x => dartBoard.GetTripleField(x.Value));
			Approvals.VerifyAll(fields, "Field");
		}

		[TestMethod]
		public void VerifyNeighboursOfFields()
		{
			var dartBoard = DartBoard.GetInstance();
			string verifyString = "";
			foreach (var field in dartBoard.Fields)
			{
				verifyString += field.ToString() + "\n";
				foreach (var neighbour in field.Neighbours)
				{
					verifyString += "\t" + neighbour.ToString() + "\n";
				}
			}
			Approvals.Verify(verifyString);
		}
	}
}
