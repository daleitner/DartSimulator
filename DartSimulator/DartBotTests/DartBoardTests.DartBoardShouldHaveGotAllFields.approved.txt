Field[0] = 0
Field[1] = 1
Field[2] = 1
Field[3] = D1
Field[4] = T1
Field[5] = 2
Field[6] = 2
Field[7] = D2
Field[8] = T2
Field[9] = 3
Field[10] = 3
Field[11] = D3
Field[12] = T3
Field[13] = 4
Field[14] = 4
Field[15] = D4
Field[16] = T4
Field[17] = 5
Field[18] = 5
Field[19] = D5
Field[20] = T5
Field[21] = 6
Field[22] = 6
Field[23] = D6
Field[24] = T6
Field[25] = 7
Field[26] = 7
Field[27] = D7
Field[28] = T7
Field[29] = 8
Field[30] = 8
Field[31] = D8
Field[32] = T8
Field[33] = 9
Field[34] = 9
Field[35] = D9
Field[36] = T9
Field[37] = 10
Field[38] = 10
Field[39] = D10
Field[40] = T10
Field[41] = 11
Field[42] = 11
Field[43] = D11
Field[44] = T11
Field[45] = 12
Field[46] = 12
Field[47] = D12
Field[48] = T12
Field[49] = 13
Field[50] = 13
Field[51] = D13
Field[52] = T13
Field[53] = 14
Field[54] = 14
Field[55] = D14
Field[56] = T14
Field[57] = 15
Field[58] = 15
Field[59] = D15
Field[60] = T15
Field[61] = 16
Field[62] = 16
Field[63] = D16
Field[64] = T16
Field[65] = 17
Field[66] = 17
Field[67] = D17
Field[68] = T17
Field[69] = 18
Field[70] = 18
Field[71] = D18
Field[72] = T18
Field[73] = 19
Field[74] = 19
Field[75] = D19
Field[76] = T19
Field[77] = 20
Field[78] = 20
Field[79] = D20
Field[80] = T20
Field[81] = 25
Field[82] = D25
